src/libraries/System.Private.CoreLib/src/System/Environment.Linux.cs

```diff
@@ -0,0 +1,14 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Runtime.InteropServices;
+using System.Text;
+
+namespace System
+{
+    public static partial class Environment
+    {
+        private static ReusableTextReader reusableReader = new ReusableTextReader();
```

Comment:
This is bad for startup time and unfriendly to IL linking:
- This will be allocated and initialized during startup since there are other statics in Environment type
- It is impossible to tree shake

Reply:
Opened #43337

Reply:
Also, this has a race condition if `Environment.WorkingSet` is used from multiple threads.

Reply:
It would be best to switch this to use the regular ArrayPool.

Reply:
Hmm, well I should have spotted that 2nd issue at least. 🤨

Reply:
Agreed that we should make it linker-friendly. However, I have not stumbled upon race condition while load testing this change. However, previously with reflection, we had a race condition for Linux which was noted here: https://github.com/dotnet/runtime/pull/43154#issuecomment-705600741. Easy to reproduce with .NET 5.0 RTM:

```c#
[Benchmark]
public long WorkingSet() => System.Environment.WorkingSet;
```

Reply:
What is the failure mode that you are seeing? I do not see anything wrong with the reflection code that this PR removed. Is it exposing bug in the reflection code or in the Process type?

Reply:
> Hmm, well I should have spotted that 2nd issue at least

@danmosemsft you would easily if we had size regression testing in CI

Reply:
Did you notice a size regression in mono with this change?

Reply:
There are size regressions every day so this is not an exception but it should be fixed

Reply:
> @danmosemsft you would easily if we had size regression testing in CI

What's the status of that?